helper/resource/testing.go

```diff
@@ -570,6 +570,14 @@ type TestStep struct {
 	// at the end of the test step that is verifying import behavior.
 	ImportStatePersist bool
 
+	//---------------------------------------------------------------
+	// RefreshState testing
+	//---------------------------------------------------------------
+
+	// RefreshState, if true, will test the functionality of `terraform
+	// refresh` by refreshing the state.
+	RefreshState bool
```

Comment:
We should consider what website documentation we can add around this. It'd also be good to denote here and in the website that its current intention is refresh, (potentially) plan, and optionally running checks.

Reply:
This field should also document that it cannot be the first `TestStep` and any other potential restrictions or contradictory definitions (such as `ImportState` + `RefreshState` for example)

Reply:
I've updated the documentation. In terms of the website docs, where do you think they should be added? The information around `ImportState` is in `/plugin/sdkv2/resources`. As we're just discussing testing here should be add a new section or add/amend an area of the website docs that already exist?

Reply:
I think https://www.terraform.io/plugin/sdkv2/testing/acceptance-tests/teststep is okay enough for now. Ideally that page would probably be broken up by test mode, but adding information in the current information architecture should be okay. 👍